src/Game/Skills.ts

```diff
@@ -49,6 +49,10 @@ export function combineEffects<T extends PlayerState>(f1: EffectFn<T>, ...fs: Ar
 	};
 }
 
+export function combinePredicatesAnd<T extends PlayerState>(f1: StatePredicate<T>, ...fs: Array<StatePredicate<T>>): StatePredicate<T> {
+	return (state: T) => f1(state) && fs.map((pred) => pred(state)).every(p => p);
```

Comment:
```suggestion
	return (state: T) => f1(state) && fs.every((pred) => pred(state));
```

I mixed this up with the analogous python function in my original suggestion, this should be better